tests/data/test_delta_dataset_source.py

```diff
@@ -113,3 +115,108 @@ def test_delta_dataset_source_too_many_inputs(spark_session, tmp_path):
 
     with pytest.raises(MlflowException, match='Must specify exactly one of "path" or "table_name"'):
         DeltaDatasetSource(path=tmp_path, delta_table_name="temp_delta_too_many_inputs")
+
+
+def test_uc_table_id_retrieval_works(spark_session, tmp_path):
+    def mock_resolve_table_name(table_name):
+        if table_name == "temp_delta_versioned_with_id":
+            return "default.temp_delta_versioned_with_id"
+        return table_name
+
+    def mock_lookup_table_id(table_name):
```

Comment:
Could we mock the `GetTable` API response instead of mocking the full method to get table name? We do want to test the new logic introduced here

Reply:
Hm, was it not possible to address this?

Reply:
Specifically I'm hoping for a test that verifies `call_endpoint` is called with the expected args; see e.g. https://github.com/mlflow/mlflow/blob/6e5ec775d1351ec3d8e88da87a317bdecb9f342d/tests/store/_unity_catalog/model_registry/test_unity_catalog_rest_store.py#L520-L593 for a similar example